Source/mpqapi.cpp

```diff
@@ -434,63 +434,57 @@ BOOL mpqapi_has_file(const char *pszName)
 
 BOOL OpenMPQ(const char *pszArchive, BOOL hidden, DWORD dwChar)
 {
-	const char *v3;         // ebp
-	BOOL v4;                // esi
-	DWORD v6;               // edi
-	int v8;                 // eax
-	int v10;                // eax
-	const char *lpFileName; // [esp+10h] [ebp-70h]
-	DWORD dwTemp;           // [esp+14h] [ebp-6Ch]
-	_FILEHEADER fhdr;       // [esp+18h] [ebp-68h]
-
-	v3 = pszArchive;
-	v4 = hidden;
-	lpFileName = pszArchive;
+	DWORD dwFlagsAndAttributes;
+	DWORD key;
+	DWORD dwTemp;
+	_FILEHEADER fhdr;
+
 	InitHash();
-	if (!mpqapi_set_hidden(v3, v4))
-		return 0;
-	v6 = (unsigned char)gbMaxPlayers > 1u ? FILE_FLAG_WRITE_THROUGH : 0;
-	save_archive_open = 0;
-	sghArchive = CreateFile(v3, GENERIC_READ | GENERIC_WRITE, 0, NULL, OPEN_EXISTING, v6, NULL);
+	if (!mpqapi_set_hidden(pszArchive, hidden)) {
+		return FALSE;
+	}
+	dwFlagsAndAttributes = gbMaxPlayers > 1 ? FILE_FLAG_WRITE_THROUGH : 0;
+	save_archive_open = FALSE;
+	sghArchive = CreateFile(pszArchive, GENERIC_READ | GENERIC_WRITE, 0, NULL, OPEN_EXISTING, dwFlagsAndAttributes, NULL);
 	if (sghArchive == INVALID_HANDLE_VALUE) {
 		sghArchive = CreateFile(lpFileName, GENERIC_READ | GENERIC_WRITE, 0, NULL, CREATE_ALWAYS, v6 | (v4 != 0 ? FILE_ATTRIBUTE_SYSTEM | FILE_ATTRIBUTE_HIDDEN : 0), NULL);

```

Comment:
```suggestion
		sghArchive = CreateFile(pszArchive, GENERIC_READ | GENERIC_WRITE, 0, NULL, CREATE_ALWAYS, v6 | (v4 != 0 ? FILE_ATTRIBUTE_SYSTEM | FILE_ATTRIBUTE_HIDDEN : 0), NULL);
```

Reply:
Oh, it looks like it happened due to rebase, sorry.